portability-spi-transfer/src/main/java/org/datatransferproject/spi/transfer/provider/ImportResult.java

```diff
@@ -47,6 +48,17 @@ public ResultType getType() {
     return throwable;
   }
 
+  /**
+   * Returns the number of imported items
+   */
+  public Map<String, Integer> getCounts() {
+    return counts;
+  }
+
+  public void setCounts(Map<String, Integer> counts) {
```

Comment:
To keep this immutable as it's returned to callers, we can either

1) Create a new object ImportResultWithCounts (with a better name) that wraps the ImportResult and adds the count data then we return ImportResultWithCounts 

or

2) We can make setCounts into a static factory method, e.g. withCounts or addCounts that returns a ImportResult which is a copy of the current one but with count data added

Both of these are to keep a clean immutable contract to callers.

Thoughts?

Reply:
I think both options make sense. I tried to implement the second one because I thought it might be slightly better to not add a new level of indirection.

The changes are:

void set counts ->(transformed to) ImportResult copyWithCounts -> which calls a new constructor to create a shallow copy of the current object but with the field counts referencing the mapping given as an argument.